code/game/objects/structures/xeno.dm

```diff
@@ -901,27 +941,39 @@ TUNNEL
 		HasProximity(A)
 
 /obj/structure/xeno/acidwell/HasProximity(atom/movable/AM)
-	if(!iscarbon(AM))
+	if(!isliving(AM))
 		return
-	var/mob/living/carbon/C = AM
-	if(C.stat == DEAD)
+	var/mob/living/stepper = AM
+	if(stepper.stat == DEAD)
 		return
 	if(!charges)
 		return
-	if(isxeno(C))
-		if(!(C.on_fire))
+
+	var/datum/effect_system/smoke_spread/xeno/acid/A
+
+	if(isxeno(stepper))
+		if(!(stepper.on_fire))
 			return
-		C.ExtinguishMob()
+		A = new(get_turf(stepper)) //spawn acid smoke when charges are actually used
+		A.set_up(0, src) //acid smoke in the immediate vicinity
+		A.start()
+		stepper.ExtinguishMob()
 		charges--
```

Comment:
flamers are a bitch, have you thought about making acid wells no longer depletes when extinguishing mob?

Reply:
Since you can recharge the wells more readily now I'd rather wait and see at this time.